packages/flutter/lib/src/widgets/scrollable.dart

```diff
@@ -761,12 +761,32 @@ class ScrollableState extends State<Scrollable> with TickerProviderStateMixin, R
     );
   }
 
-  // Returns the delta that should result from applying [event] with axis and
-  // direction taken into account.
+  // Returns the delta that should result from applying [event] with axis,
+  // direction, and any modifiers specified by the ScrollBehavior taken into
+  // account.
   double _pointerSignalEventDelta(PointerScrollEvent event) {
-    double delta = widget.axis == Axis.horizontal
-      ? event.scrollDelta.dx
-      : event.scrollDelta.dy;
+    late double delta;
+    final Set<LogicalKeyboardKey> pressed = HardwareKeyboard.instance.logicalKeysPressed;
+    final bool flipAxes = pressed.any(_configuration.pointerAxisModifiers.contains)
+      // Axes are only flipped for physical mouse wheel input.
+      // On some platforms, like web, trackpad input is handled through pointer
+      // signals, but should not be included in this axis modifying behavior.
+      // This is because on a trackpad, all directional axes are available to
+      // the user, while mouse scroll wheels typically are restricted to one
+      // axis.
+      && event.kind == PointerDeviceKind.mouse;
```

Comment:
(uber, maybe not even true, nit): Don't we usually put the `&&` at the end of the line above, except when they're stacked in a return statement?

Reply:
Good question, I don't know. I have seen it both ways. I think I figured `&&` on the preceding line exceeded the line limit so I placed it below. But I'm fine either way. Just moved it up. :)